satpy/composites/glm.py

```diff
@@ -0,0 +1,121 @@
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Composite classes for the GLM instrument."""
+
+import logging
+
+import xarray as xr
+from satpy.composites import GenericCompositor
+from satpy.writers import get_enhanced_image
+
+LOG = logging.getLogger(__name__)
+
+
+class HighlightCompositor(GenericCompositor):
+    """Highlight pixels of a layer by an amount determined by a secondary layer.
+
+    The highlighting is applied per channel to either add or subtract an
+    intensity from the primary image. In the addition case, the code is
+    essentially doing::
+
+        highlight_factor = (highlight_data - min_highlight) / (max_highlight - min_highlight)
+        channel_result = primary_data + highlight_factor * max_factor
+
+    See the ``highlight_channel`` option to control if the effect is additive,
+    subtractive, or not applied at all.
+
+    """
+
+    def __init__(self, name, min_highlight=0.0, max_highlight=10.0, max_factor=0.8,
+                 highlight_channel=(True, True, False, None), **kwargs):
+        """Initialize composite with highlight factor options.
+
+        Args:
+            min_highlight (float): Minimum raw value of the "highlight" data
+                that will be used for linearly scaling the data along with
+                ``max_hightlight``.
+            max_highlight (float): Maximum raw value of the "highlight" data
+                that will be used for linearly scaling the data along with
+                ``min_hightlight``.
+            max_factor (float): Maximum effect that the highlight data can
+                have on the primary image data. This will be multiplied by
+                the linearly scaled highlight data and then added or
+                subtracted from the highlight channels. See class docstring
+                for more information.
+            highlight_channel (tuple): Series of booleans or None for every
+                channel in the RGBA image (4). True means apply the highlight
+                effect by adding, False means apply the highlight effect by
+                subtracting, and None means don't apply the highlight. By
+                default this will add to the Red and Green channels, subtract
+                from the Blue channel, and not effect the Alpha channel. This
+                results in yellow highlights in the resulting image.
```

Comment:
I think I understand how this works, but I don't like the way `highlight_channel` is used: `True` for additive, `False` for substractive is kind of strange.
How about combining `max_factor` and `highlight_channel` in one argument? That would also allow more control over the resulting color. Like `highlight_channel = (.8, .6, -.8, 0)` for a more orange look.

Reply:
Interesting idea. I'm ok with that. The `highlight_channel` argument came up as a last minute "this would be more flexible" thing so I'm happy to change `max_factor` to be a tuple/list as that should make more sense.